### PR TITLE
Add functionality to search by birthday month

### LINKED ISSUE #120: 📖 As a welfare IC, I can search for contacts based on birthday month
...so that I can plan birthday parties/events on a monthly-basis.

________________________________________________________________
### Describe the feature
*Define precisely the desired behavior.*

Overload `list` to include filtering by birthday month.
Possible formats to support:
- `list -b jan`
- `list -b Jan`
- `list -b January`
- `list -b 1`

As a stretch goal, can extend to multiple months, space-demarcated, e.g. `list -b jan apr`

________________________________________________________________
### Additional context
*What is the current implementation, and the required scope of work?*

Currently no direct birthday month extraction and comparison.

### PR DESCRIPTION
**Pull request by pyuxiang**
### Important note

Extension of #180, do not review until PR merged.

### Feature changes

Fixes #120 (without the multiple birthday in same prefix).

### Test syntax

`list [...] -b 1` (alias: `Jan`, `January`, case-insensitive)

image

### CODE REVIEW

File: src/main/java/seedu/partyplanet/logic/parser/EListCommandParser.java [hunk 1]
```diff
@@ -140,7 +144,7 @@ public EListCommand parse(String args) throws ParseException {
     private Comparator<Event> applySortDirection(
             Comparator<Event> comparator, ArgumentMultimap argMap) throws ParseException {
         Optional<String> orderType = argMap.getValue(PREFIX_ORDER);
-        if (orderType.isEmpty()) {
+        if (orderType.isEmpty() || comparator == SORT_EVENTDATE_UPCOMING) {
```
Comment (garyljj) [inline src/main/java/seedu/partyplanet/logic/parser/EListCommandParser.java:147]: This makes sort_upcoming unable to work with asc/desc.
Isit intented? If so then just remember to document
Same for list

**Review by garyljj — COMMENTED** [review]
Tested on GUI, works well
Just the following issue

File: src/main/java/seedu/partyplanet/logic/parser/EListCommandParser.java (hunk 1 above)
Comment (garyljj) [reply src/main/java/seedu/partyplanet/logic/parser/EListCommandParser.java:147]: Oops comment was meant for PR 180, shifted over

**Review by garyljj — APPROVED** [review]
Tested, LGTM

**Review by Ellevy — APPROVED** [review]
LGTM

**Pull request merged**